File: problems/custom/python_problems/custom2.py
def custom2():
    """
    This is a toy example to show how to construct a .py file that returns a dictionary that describes a bound-constrained optimization problem.
    """
    p_dict = {}
    p_dict['name'] = 'custom2'
    p_dict['x0'] = [1.0, 1.0]
    p_dict['fun'] = fun
    p_dict['xl'] = [-5.0, -5.0]
    p_dict['xu'] = [5.0, 5.0]
    
    return p_dict

def fun(x):
    return sum(xi**2 for xi in x)

File: matlab/optiprofiler/problem_libs/custom/matlab_problems/custom2.m
function problem = custom2()
    % This is a toy example to show how to construct a .m file that returns
    % a instance of Problem class representing a bound-constrained problem.

    p_struct.name = 'custom2';
    p_struct.x0 = [1; 1];
    p_struct.fun = @fun;
    p_struct.xl = [-5; -5];
    p_struct.xu = [5; 5];

    problem = Problem(p_struct);
end

function f = fun(x)
    f = sum(x.^2);
end